packages/marshal/test/test-marshal-smallcaps.js

```diff
@@ -0,0 +1,327 @@
+ *  * `$` - remotable
+ *  * `&` - promise
+ */
+test('smallcaps encoding examples', t => {
+  const { serialize } = makeMarshal(() => 'slot', undefined, {
+    errorTagging: 'off',
+    useSmallcaps: true,
+  });
+  const assertSer = (val, body, slots, message) =>
+    t.deepEqual(serialize(val), { body, slots }, message);
+
+  // Numbers
+  assertSer(0, '#0', [], 'zero');
+  assertSer(500n, '#"+500"', [], 'bigint');
+  assertSer(-400n, '#"-400"', [], '-bigint');
+
+  // Constants
+  assertSer(NaN, '#"#NaN"', [], 'NaN');
+  assertSer(Infinity, '#"#Infinity"', [], 'Infinity');
+  assertSer(-Infinity, '#"#-Infinity"', [], '-Infinity');
+  assertSer(undefined, '#"#undefined"', [], 'undefined');
+
+  // Strings
+  assertSer('unescaped', '#"unescaped"', [], 'unescaped');
+  assertSer('#escaped', `#"'#escaped"`, [], 'escaped #');
+  assertSer('+escaped', `#"'+escaped"`, [], 'escaped +');
+  assertSer('-escaped', `#"'-escaped"`, [], 'escaped -');
+  assertSer('%escaped', `#"'%escaped"`, [], 'escaped %');
+
+  // Symbols
+  assertSer(Symbol.iterator, '#"%@@iterator"', [], 'well known symbol');
+  assertSer(Symbol.for('foo'), '#"%foo"', [], 'reg symbol');
+  assertSer(
+    Symbol.for('@@foo'),
+    '#"%@@@@foo"',
+    [],
+    'reg symbol that looks well known',
+  );
+
+  // Remotables
+  const foo = Far('foo', {});
+  const bar = Far('bar', {});
+  assertSer(foo, '#"$0.Alleged: foo"', ['slot'], 'Remotable object');
+  assertSer(
+    harden([foo, bar, foo]),
+    '#["$0.Alleged: foo","$1.Alleged: bar","$0"]',
+    ['slot', 'slot'],
+    'Only show iface once',
+  );
+
+  // Promises
+  const p = harden(Promise.resolve(null));
+  assertSer(p, '#"&0"', ['slot'], 'Promise');
+
+  // Arrays
+  assertSer(harden([1, 2n]), '#[1,"+2"]', [], 'array');
+
+  // Records
+  assertSer(harden({ foo: 1, bar: 2n }), '#{"bar":"+2","foo":1}', [], 'record');
+
+  // Tagged
+  const tagged = makeTagged('foo', 'bar');
+  assertSer(tagged, '#{"#tag":"foo","payload":"bar"}', [], 'tagged');
+
+  // Error
+  const err = harden(URIError('bad uri'));
+
+  // non-passable errors alone still serialize
+  assertSer(err, '#{"#error":"bad uri","name":"URIError"}', [], 'error');
+  const nonPassableErr = Error('foo');
+  // @ts-expect-error this type error is what we're testing
+  nonPassableErr.extraProperty = 'something bad';
+  harden(nonPassableErr);
+  t.throws(() => passStyleOf(nonPassableErr), {
+    message:
+      'Passed Error has extra unpassed properties {"extraProperty":{"value":"something bad","writable":false,"enumerable":true,"configurable":false}}',
+  });
+  assertSer(
+    nonPassableErr,
+    '#{"#error":"foo","name":"Error"}',
+    [],
+    'non passable errors pass',
+  );
+
+  // Hilbert record
+  assertSer(
+    harden({
+      '#tag': 'what',
+      '#error': 'me',
+      '#huh': 'worry',
+      '': 'empty',
+      '%sym': 'not a symbol',
+    }),
+    '#{"":"empty","\'#error":"me","\'#huh":"worry","\'#tag":"what","\'%sym":"not a symbol"}',
```

Comment:
Oh I didn't realize that `'` was getting escaped in JSON encoding. That's unfortunate...

Reply:
I'm an idiot, this is because the test uses ` ' ` for the outer quotes. Please update the test to use `` ` `` instead.

Reply:
I think your looking at the wrong metalevel. To print the string as a single-quote quoted string, the embedded single quotes are escaped. But they're not actually escaped in JSON string. In any case, I expect these hilbert cases to be rare enough that it shouldn't make much difference. Resolving.

Reply:
Yes I read it wrong, but I think to avoid confusion, we can very well use a template / backticks in this case. Reopened.

Reply:
I had meant to quote the expected value with backticks to avoid escapes, but I think I do find the bang more readable as an escape anyway.

```js
    `#{"":"empty","'#error":"me","'#huh":"worry","'#tag":"what","'%sym":"not a symbol"}`,
```